lib/Alchemy/Phrasea/Command/Plugin/DownloadPlugin.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * This file is part of Phraseanet
+ *
+ * (c) 2005-2016 Alchemy
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Alchemy\Phrasea\Command\Plugin;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\ArrayInput;
+
+use Cz\Git\GitRepository as GitRepository;
+
+
+
+class DownloadPlugin extends AbstractPluginCommand
+{
+
+    public function __construct()
+    {
+        parent::__construct('plugins:download');
+
+        $this
+            ->setDescription('Downloads a plugin to Phraseanet')
+            ->addArgument('source', InputArgument::REQUIRED, 'The source is a remote url (.zip or .git)')
+            ->addArgument('destination', InputArgument::OPTIONAL, 'Download destination');
+    }
+
+
+    protected function doExecutePluginAction(InputInterface $input, OutputInterface $output)
+    {
+
+        $source = $input->getArgument('source');
+        $destination = $input->getArgument('destination');
+
+        $destination_subdir = '/plugin-'.time();
```

Comment:
you can have conflict with `time()` (if many plugins are installed automatically at the same time for instance).
You can `md5()` the source.